src/instanceMethods/_main.js

```diff
@@ -51,6 +43,24 @@ export function _main (userParams) {
   return swalPromise(this, domCache, innerParams)
 }
 
+const cleanUpPreviousPopup = () => {
+  // Check if there is another Swal closing
+  if (dom.getPopup() && globalState.swalCloseEventFinishedCallback) {
+    globalState.swalCloseEventFinishedCallback()
+    delete globalState.swalCloseEventFinishedCallback
+  }
+
+  // Check if there is a swal disposal defer timer
+  if (globalState.deferDisposalTimer) {
+    clearTimeout(globalState.deferDisposalTimer)
+    delete globalState.deferDisposalTimer
+  }
```

Comment:
Maybe this logic should be organized under the `_destroy` method (so that it's easier to notice and/or locate). Also, then we have the option of cleaning up a bit (no need to pass all these variables through `globalState`, since we have access to `this` inside `_destroy`) but I'm not sure if that is necessary.

Reply:
Referring to the above *two* blocks of code

Reply:
Good point, done :+1: